File: webservices/src/main/java/fieg/modulos/boleto/dto/ConsultaDadosBoletoCR5/DadoBoletoCr5DTO.java
package fieg.modulos.boleto.dto.ConsultaDadosBoletoCR5;

import fieg.modulos.boleto.enums.SituacaoBoleto;
import fieg.modulos.cobrancaagrupada.enums.SituacaoCobrancaAgrupada;
import fieg.modulos.cobrancacliente.enums.SituacaoCobrancaCliente;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class DadoBoletoCr5DTO {
    private Integer idBoleto;
    private String nossoNumero;
    private SituacaoBoleto bolSituacao;
    private BigDecimal bolValor;
    private BigDecimal bolDesconto;
    private BigDecimal bolJuros;
    private BigDecimal bolMulta;
    private BigDecimal bolPago;
    private LocalDateTime bolVencimento;
    private LocalDateTime bolDataPagamento;
    private LocalDateTime bolDataCredito;
    private LocalDateTime bolEmissao;
    private LocalDateTime bolCancelamento;
    private Integer idCobrancasCliente;
    private Integer idCobrancasAgrupada;
    private String operadorInclusao;
    private String operadorAlteracao;
    private String operadorCancelamento;
    private SituacaoCobrancaAgrupada grupoStatus;
    private Integer idUnidade;
    private Integer codUnidade;
    private String cnpjCedente;
    private String codigoBeneficiario;
    private Integer idArquivosRetornos;
    private Integer idArquivosDetalhes;
    private String bolRemessa;
}

File: web/src/models/DTOs/ConsultaBoletoCaixa/DadoBoletoCr5DTO.ts
import { SituacaoAgrupado } from "../CobrancasAgrupadas/SituacaoAgrupado";
import { SituacaoBoleto } from "./SituacaoBoleto";

export interface DadoBoletoCr5DTO {
    idBoleto: number;
    nossoNumero: string;
    bolSituacao: SituacaoBoleto;
    bolValor: number;
    bolDesconto: number | null;
    bolJuros: number | null;
    bolMulta: number | null;
    bolPago: number | null;
    bolVencimento: string; // formato ISO: 'YYYY-MM-DD'
    bolDataPagamento: string | null;
    bolDataCredito: string | null;
    bolEmissao: string | null;
    bolCancelamento: string | null;
    idCobrancasCliente: number;
    idCobrancasAgrupada: number;
    operadorInclusao: string;
    operadorAlteracao: string | null;
    operadorCancelamento: string | null;
    grupoStatus: SituacaoAgrupado;
    idUnidade: number;
    codUnidade: number;
    cnpjCedente: string;
    codigoBeneficiario: string;
    idArquivosRetornos: number | null;
    idArquivosDetalhes: number | null;
    bolRemessa: string | null;
}